--- infiniband-diags-1.6.5/src/rdma-ndd.c.orig	2015-01-28 20:12:33.000000000 +0100
+++ infiniband-diags-1.6.5/src/rdma-ndd.c	2015-02-15 21:04:04.865070220 +0100
@@ -146,7 +146,7 @@
 		syslog(LOG_INFO, "%s: change (%s) -> (%s)\n",
 			device, nd, new_nd);
 		rewind(f);
-		fprintf(f, new_nd);
+		fputs(new_nd, f);
 	}
 
 	rc = 0;
@@ -244,7 +244,7 @@
 			file, line, fn);
 	if (off < MSG_MAX-1)
 		vsnprintf(msg+off, MSG_MAX-off, format, args);
-	syslog(LOG_ERR, msg);
+	syslog(LOG_ERR, "%s", msg);
 }
 
 static void setup_udev(void)
